_posts/en/newsletters/2020-06-10-newsletter.md

```diff
@@ -0,0 +1,393 @@
+  transaction or to maintain a database of verified UTXO data.
+
+    Hardware wallets don't maintain a UTXO set, so the only trustless
+    way for them to determine the amount of fee paid by a transaction
+    with legacy UTXOs is to get a copy of each legacy UTXO's previous
+    transaction, hash the previous transaction to ensure its txid
+    matches the UTXO reference, and use the now-verified amount of the
+    UTXO to perform the fee calculation.  In the worst case, legacy transactions can
+    be almost as large as one megabyte and it's possible for a spending
+    transaction to reference thousands of previous transactions, so amount
+    verification can require the processing of gigabytes of data by
+    resource-constrained hardware wallets.
+
+    One of the several improvements made in [BIP143][] segwit v0 was an
+    attempt to eliminate this burden by having [signatures commit to the
+    amount][bip143 motivation] of the UTXO they were spending.  This
+    meant that any signatures that committed to an incorrect amount
+    would be invalid, which developers of both [Bitcoin Core][towns
+    benefits] and [hardware wallets][palatinus inpatient] believed would
+    allow signers to safely accept amounts from untrusted software.
+
+    Unfortunately, signing a single UTXO's amount turned out to be
+    insufficient.  In 2017, Greg Sanders [described][sanders attack] an
+    attack that can be used to trick a user into significantly
+    overpaying their transaction fee.  To start, an attacker takes
+    control of the desktop or mobile software that interfaces with a
+    hardware wallet and waits for the user to initiate a payment.  The
+    compromised software creates two variations of the requested
+    transaction that each spend the same two segwit UTXOs controlled by
+    the user's attached hardware wallet.  In the first transaction, the
+    compromised software understates the amount of one of the UTXOs
+    being spent.  This will cause the hardware wallet to under-calculate
+    the transaction fee by that same understated amount.  In the second
+    transaction, the compromised software understates the amount of the
+    second UTXO.  That will cause the hardware wallet to also
+    under-calculate the second transaction's fee.
+
+    The compromised software sends the first transaction to the hardware
+    wallet.  The user checks the amount, receiver address, and calculated
+    transaction fee.  The user authorizes signing and the signed
+    transaction is returned to the compromised software.  This first
+    transaction is invalid under BIP143 because one of its signatures
+    commits to an incorrect UTXO amount.  The compromised software then
+    claims that there was some minor problem that prevented the
+    transaction from being broadcast and tells the user they need to
+    re-sign the transaction, but the wallet doesn't re-send the first
```

Comment:
s/wallet/software/. You've distinguished between 'hardware wallet' and 'compromised software' up to this point. Using the word wallet here is confusing.